src/io/perun/core.clj

```diff
@@ -11,15 +12,15 @@
   "Return metadata on files. Files metadata is a list.
    Internally it's stored as a map indexed by `:path`"
   [fileset]
-  (-> fileset meta +meta-key+ vals))
+  (keep +meta-key+ (vals (:tree fileset))))
 
 (defn key-meta [data]
   (into {} (for [d data] [(:path d) d])))
 
 (defn set-meta
-  "Update `+meta-key+` metadata for the fileset and return updates fileset"
+  "Update `+meta-key+` metadata for files in `data` and return updated fileset"
   [fileset data]
-  (vary-meta fileset assoc +meta-key+ (key-meta data)))
+  (boot/add-meta fileset (into {} (for [d data] [(:path d) {+meta-key+ d}]))))
```

Comment:
actually, on a second look, I think the function was just moved in this commit.  Here is the real commit: https://github.com/boot-clj/boot/commit/038d8dde09704e3d22b5cdfdd8be984825f04bf9